### DISCUSSION TITLE
Adding additional metadata to tracks and writing them to id3 tags?

### DISCUSSION THREAD

**Discussion by Neurrone** [beetbox/beets#4271] [Q&A] [answered]
Hi,

I'd like to be able to set the following ID3 tags on tracks: albumartist (as a different value from artist), comment, contentgroup, description, itunesmediatype, movementname, movement, releasetime, series, series-part, subtitle.

1. I don't see mappings for these fields in `TrackInfo` and was wondering if it would be possible for me to still have these tags written on import?
2. Is it possible for me to store additional fields on TrackInfo and AlbumInfo on import, beyond the built-in fields?

**Comment by sampsyo** [comment]
The answer about `TrackInfo` and `AlbumInfo` is "yes"! The fixed list of fields there is actually just a legacy thing; you can put anything you like in the keyword arguments (i.e., just pass it along like `movementname='foo'`).

To convey these to ID3 tags, you will need to add a tag mapping, as described here:
https://beets.readthedocs.io/en/stable/dev/plugins.html#extend-mediafile

**Comment by Neurrone** [comment, marked as answer]
Thanks, I got that to work.

1. I realize that although I want to write additional tags to all chapters / tracks in the album, I may not want to store them in the database to avoid duplicate information.
    
    For example, its probably fine to store genre for each track even though its the same as for the overall album, but storing the description / summary of the book both on an album and on a per-track basis would lead to a lot of duplicate info in the database.
    
    Is there a way for me to write additional ID3 tags to each individual file without having to save those tags to the database?
2. Something else I also noticed is when setting the comments field for `TrackInfo`, I noticed it got truncated when writing it to the file. I'm wondering if Beets does any sort of truncation on comments?
3. Where's the mapping that Beets uses internally between its own TrackInfo fields to ID3 tag names?

**Reply by sampsyo** [reply]
Writing tags to the files without putting them in the database is a little funky and not exactly the way beets was designed. The idea is that the database is a 1:1 reflection of what's in the files. It is probably possible with some hacking (i.e., directly accessing the MediaFile for the path under the hood), I would start by not worrying too much about the duplication—that's the way that most album-level tags work already.

I'm not aware of any truncation that goes on intentionally, no.

The mapping to tag names is in [MediaFile](https://github.com/beetbox/mediafile) (see the listings of MediaField declarations near the bottom of the source file).

**Reply by Neurrone** [reply]
Thanks, I got both custom tags via item fields to work and I believe I also got a tag to be written without the corresponding item field.

```python
# plugin constructor
itunes_media_type = mediafile.MediaField(
    mediafile.MP3DescStorageStyle(u'ITUNESMEDIATYPE'),
    mediafile.StorageStyle(u'ITUNESMEDIATYPE')
)
self.add_media_field('itunes_media_type', itunes_media_type)

def on_write(self, item, path, tags):
        tags["itunes_media_type"] = "Audiobook"
```

I'll also not worry about the duplication for now.
